projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx

```diff
@@ -62,118 +17,141 @@ export class DotFormComponent {
      * @param CustomEvent event
      * @memberof DotFormComponent
      */
     @Listen('valueChange')
     onValueChange(event: CustomEvent): void {
-        const { name, value, fieldType } = event.detail;
-        this.value[name] = fieldType ? fieldParamsConversionToBE[fieldType](value) : value;
-        this.updateFieldsValues();
+        const { tagName } = event.target as HTMLElement;
+        const { name, value } = event.detail;
+        const transform = fieldParamsConversionToBE[tagName];
+
+        this.value[name] = transform ? transform(value) : value;
+        this.fields = this.getUpdateFieldsValues();
     }
 
     /**
-     * Listen for "statusChange" and updates the form status with new value.
+     * Update the form status when statusChange in any of the child fields
      *
      * @param CustomEvent event
      * @memberof DotFormComponent
      */
     @Listen('statusChange')
-    onStatusChange(event: CustomEvent): void {
-        this.fieldsStatus[event.detail.name] = event.detail.status;
+    onStatusChange({ detail }: CustomEvent): void {
+        this.fieldsStatus[detail.name] = detail.status;
+
         this.status = updateStatus(this.status, {
-            dotTouched: this.getStatusValue('dotTouched'),
-            dotPristine: this.getStatusValue('dotPristine'),
-            dotValid: this.getStatusValue('dotValid')
+            dotTouched: this.getTouched(),
+            dotPristine: this.getStatusValueByName('dotPristine'),
+            dotValid: this.getStatusValueByName('dotValid')
         });
     }
 
+    @Watch('fields')
+    fieldsWatch() {
+        this.updateValue();
+    }
+
+    @Watch('fieldsToShow')
+    fieldsToShowWatch() {
+        this.updateValue();
+    }
+
     hostData() {
         return {
             class: getClassNames(this.status, this.status.dotValid)
         };
     }
 
     componentWillLoad() {
-        this.fields.forEach((field: DotCMSContentTypeField) => {
-            if (this.getFieldTag(field)) {
-                this.value[field.variable] = field.defaultValue || '';
-            }
-        });
+        this.updateValue();
     }
 
     render() {
         return (
-            <form onSubmit={(evt: Event) => this.handleSubmit(evt)}>
-                {this.fields.map((field: DotCMSContentTypeField) => this.getField(field))}
+            <form onSubmit={this.handleSubmit.bind(this)}>
+                <div class="form__fields">
+                    {this.fields.map((field: DotCMSContentTypeField) => this.getField(field))}
+                </div>
+
                 <div class="form__buttons">
-                    <button type="button" onClick={() => this.resetForm()}>
+                    <button type="reset" onClick={() => this.resetForm()}>
                         {this.resetLabel}
                     </button>
-                    <button type="submit" disabled={!this.status.dotValid || null}>
+                    <button type="submit" disabled={!this.status.dotValid}>
```

Comment:
is this `ok`? I mean when it is `disabled=false` does the button works? or it is still disabled?

Reply:
Yes... you can see the button getting enabled/disabled based on the property of the form